lib/components/narrative/access-leg.js

```diff
@@ -12,48 +12,58 @@ export default class AccessLeg extends Component {
     setActiveLeg: PropTypes.func,
     setActiveStep: PropTypes.func
   }
+
   _onLegClick (e, leg, index) {
     if (this.props.active) {
       this.props.setActiveLeg(null)
     } else {
       this.props.setActiveLeg(index, leg)
     }
   }
+
   _onStepClick (e, step, index) {
     if (index === this.props.activeStep) {
       this.props.setActiveStep(null)
     } else {
       this.props.setActiveStep(index, step)
     }
   }
+
   render () {
     const { active, activeStep, index, leg } = this.props
+    const legClickHandler = (e) => this._onLegClick(e, leg, index)
```

Comment:
`leg` and `index` are retrieved from the props.

```js
  legClickHandler = (e) => this._onLegClick(e, this.props.leg, this.props.index)
```

Should work the same and not be re-created on each render. It's not going to be a serious performance hit here, but it's good practice.